apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,218 @@
+	GitHubPhaseReady GitHubIdentityProviderPhase = "Ready"
+
+	// GitHubPhaseError is the phase for an GitHubIdentityProvider in an unhealthy state.
+	GitHubPhaseError GitHubIdentityProviderPhase = "Error"
+)
+
+type GitHubAllowAllUsersToLoginSpec string
+
+const (
+	// GitHubAllowAllUsersToLogin means any GitHub user is allowed to log in to this Pinniped,
+	// regardless of their organization membership or lack thereof.
+	GitHubAllowAllUsersToLogin GitHubAllowAllUsersToLoginSpec = "Allow"
+
+	// GitHubDoNotAllowAllUsersToLogin means only those users with membership in specified GitHub
+	// organizations are allowed to log in.
+	GitHubDoNotAllowAllUsersToLogin GitHubAllowAllUsersToLoginSpec = "Deny"
+)
+
+// GitHubIdentityProviderStatus is the status of an GitHub identity provider.
+type GitHubIdentityProviderStatus struct {
+	// Phase summarizes the overall status of the GitHubIdentityProvider.
+	//
+	// +kubebuilder:default=Pending
+	// +kubebuilder:validation:Enum=Pending;Ready;Error
+	Phase GitHubIdentityProviderPhase `json:"phase,omitempty"`
+
+	// Conditions represents the observations of an identity provider's current state.
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// GitHubApiConfig allows configuration for GitHub Enterprise Server
+type GitHubApiConfig struct {
+	// Host is required only for GitHub Enterprise Server.
+	// Defaults to using GitHub's public API (github.com).
+	// +optional
+	Host string `json:"host,omitempty"`
+
+	// TLS configuration for GitHub Enterprise Server.
+	// +optional
+	TLS *TLSSpec `json:"tls,omitempty"`
+}
+
+// GitHubUsernameClaimSpec allows the user to specify which attribute(s) from GitHub to use for the downstream username.
+// See the response for https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user.
+type GitHubUsernameClaimSpec string
+
+const (
+	// GitHubUsernameId specifies using the `id` attribute from GitHub as the downstream username.
+	GitHubUsernameId GitHubUsernameClaimSpec = "id"
+
+	// GitHubUsernameLogin specifies using the `login` attribute from GitHub as the downstream username.
+	GitHubUsernameLogin GitHubUsernameClaimSpec = "login"
+
+	// GitHubUsernameLoginAndId specifies combining the `login` and `id` attributes from GitHub as the
+	// downstream username, separated by a colon. Example: "my-login:1234"
+	GitHubUsernameLoginAndId GitHubUsernameClaimSpec = "login:id"
+)
+
+// GitHubGroupNamesSpec allows the user to specify which attribute from GitHub to use for the downstream groups.
+// See the response for https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams-for-the-authenticated-user.
+type GitHubGroupNamesSpec string
+
+const (
+	// GitHubGroupsAsNames specifies using the `name` attribute from GitHub as the downstream group names
```

Comment:
The singular to plural shift in the comment is a little weird, also the github docs reference "teams".  Should we make sure this is clear?

Reply:
done.